temporian/implementation/numpy/data/dtype_normalization.py

```diff
@@ -187,40 +192,20 @@ def normalize_timestamps(
 
         return values, False
 
-    if values.dtype.type in [np.str_, np.bytes_]:
+    if values.dtype.type in [np.str_, np.bytes_, np.object_]:
+        # Raises ValueError if cannot parse a value
         values = values.astype("datetime64[ns]")
 
-    if values.dtype.type == np.object_:
-        if all(isinstance(x, str) for x in values) or all(
-            isinstance(x, datetime.datetime) for x in values
-        ):
-            # values is a date. Cast to unix epoch in float64 seconds.
-            values = values.astype("datetime64[ns]")
-
-        elif all(
-            str(type(x)) == "<class 'pandas._libs.tslibs.timestamps.Timestamp'>"
-            for x in values
-        ):
-            values = np.array([x.to_numpy() for x in values])
-
     if values.dtype.type == np.datetime64:
         # values is a date. Cast to unix epoch in float64 seconds.
         values = datetime64_array_to_float64(values)
+        if np.any(values < MIN_TIMESTAMP_S):
```

Comment:
This might cause some issues (We have datasets where the unix timestamp is negative).
Could this be checked differently?

Reply:
Actually this `MIN_TIMESTAMP_S` is a negative number. It's kind of arbitrarily set to `1900-01-01`, which seemed like a safe date span 😄